src/basics.md

```diff
@@ -1770,6 +1772,110 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-impl-debug]: #ex-impl-debug
+<a name="ex-impl-debug"></a>
+## Implementing Debug for a custom type
+
+[![std-badge]][std] [![cat-os-badge]][cat-os]
+
+Manually implements the [`Debug`] trait for a custom enum and it's attributes.
+
+```rust
+use std::fmt;
+
+enum FileType {
+    Image {
+        format: String,
+        width: i32,
+        height: i32,
+    },
+    Text {
+        format: String,
+    },
+    Binary {
+        arch: String,
+    },
+}
+
+impl fmt::Debug for FileType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            FileType::Image {
+                format,
+                width,
+                height,
+            } => write!(
+                f,
+                "Image {{ format: {}, width: {}, height: {} }}",
+                format, width, height
+            ),
+            FileType::Text { format } => write!(f, "Text {{ format: {} }}", format),
```

Comment:
We actually don't need to manually write out the `Text { format: value }` struct-like format here. There are methods on the `fmt::Formatter` like [`debug_struct`](https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_struct) that will do this for us and keep it consistent with what you get from `#[derive(Debug)]`.